### PR TITLE
Add `jekyll-minifier`

### PR DESCRIPTION
**Pull request by rohandebsarkar**
[`jekyll-minifier`](https://github.com/digitalsparky/jekyll-minifier) provides a better way to minify HTML, CSS and JS files.

It reduces the file sizes by a considerable amount, resulting in faster loading of the webpages.

### CODE REVIEW

File: bin/deploy [hunk 1]
```diff
@@ -91,6 +91,9 @@ then
 fi
 git checkout -b $DEPLOY_BRANCH
 
+# Export JEKYLL_ENV=production
+export JEKYLL_ENV=production
```
Comment (alshedivat) [inline bin/deploy:95]: why is this necessary?

**Comment by alshedivat** [comment]
super cool! thanks for adding `jekyll-minifier`.  do you know if this also solves issues with having to add white spaces to ensure the rendered html looks nicely formatted? (see relevant discussion here: https://github.com/alshedivat/al-folio/pull/481#issuecomment-1001252779.)

File: bin/deploy (hunk 1 above)
Comment (rohandebsarkar) [reply bin/deploy:95]: `jekyll-minifier` only works when `JEKYLL_ENV=production` is set.

This tells the plugin that this site is build for production, that way this won't work during testing but will work while deploying. This will save a lot of time during testing, as it takes a bit of time to minify all the files.

**Comment by rohandebsarkar** [comment]
> super cool! thanks for adding `jekyll-minifier`. do you know if this also solves issues with having to add white spaces to ensure the rendered html looks nicely formatted? (see relevant discussion here: [#481 (comment)](https://github.com/alshedivat/al-folio/pull/481#issuecomment-1001252779).)

This plugin will not format the code to look good. It will delete all the whitespace and newline characters and also remove the comments. This will reduce the file size by a lot, but the code after being minified will not be in human-readable form. Said that, the browsers will auto format the code upon receiving, therefore it will not be a problem while inspecting the code. Moreover, this will only run while deploying and not while testing.
